lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -690,14 +690,16 @@ function dosomething_reportback_save($values) {
   }
   else {
     $entity = entity_load_single('reportback', $rbid);
+    $reportback_item = entity_load_single('reportback_item', $values['fid']);
   }
   try {
     // Set entity properties.
     dosomething_reportback_set_properties($entity, $values);
     // Set entity files.
-    dosomething_reportback_set_files($entity, $values);
-    // Save the entity.
+    dosomething_reportback_set_files($reportback_item, $values);
```

Comment:
well I _think_ (keep in mind i have not tested this at all) you'll want to leave that entity->save
and have a new one, that saves the files as well.